Revert #6271 from charris/change-deprecated-indexes-to-error

charris opened the pull request:
This reverts commit 0b0206c1617841ed2e5324de752ee8ede2cce791, reversing
changes made to 438cdd3d75a0bb606e7ab7f96e59744c9b78d748.

This reverts all the changes of indexing deprecation warnings to errors
that were done prior to branching Numpy 1.11.x. It is easier to make a
clean reversion here than to pick out those bits involving integer
indexing specifically. As the aim of this reversion is to give
downstream projects a bit more time to adapt I don't think the slight
overkill is worth worrying about, especially if downstream projects
start paying close attention to deprecations.

The 1.11.0-notes also need updating, but I will submit a separate PR
against master for that.

Closes #7162.

charris committed:
Revert #6271 from charris/change-deprecated-indexes-to-error

This reverts commit 0b0206c1617841ed2e5324de752ee8ede2cce791, reversing
changes made to 438cdd3d75a0bb606e7ab7f96e59744c9b78d748.

This reverts all the changes of indexing deprecation warnings to errors
that were done prior to branching Numpy 1.11.x. It is easier to make a
clean reversion here than to pick out those bits involving integer
indexing specifically. As the aim of this reversion is to give
downstream projects a bit more time to adapt I don't think the slight
overkill is worth worrying about, especially if downstream projects
start paying close attention to deprecations.

The 1.11.0-notes also need updating, but I will submit a separate PR
against master for that.

Closes #7162.

njsmith commented:
Are you planning also to bump these up to VisibleDeprecationWarnings?

charris commented:
Hmm, I wasn't. I had the impression that all that was needed was a bit of time for statsmodels and other slow releasers to make a release. I'm also hoping more people will begin testing against master as pandas does.

rgommers commented:
I think we should make them visible, or we'll have the exact same issue next release.

charris commented:
I suppose the correct thing to do here if import `VisibleDeprecationWarning` from numpy so that there is only one that can be used by everyone. Could make a function in `numpy/src/private` that does that. Do we want to keep that name?

njsmith commented:
Let's keep the name for now... We should probably do a more comprehensive overhaul of our deprecation policy, but for 1.11 minimal code changes are good :-)

charris committed:
MAINT: Use VisibleDeprecationWarning for indexing deprecations.

The Usual DeprecationWarning is considered too easy to ignore, so
indexing is not visible to some downstream packages. Use a
VisibleDeprecationWarning instead.

charris commented:
Updated.
